endpoint/endpoint_core.c

```diff
@@ -122,11 +122,32 @@ status_t send_transaction_information(const char* host, const char* port, const
     return SC_ENDPOINT_SEND_TRANSFER;
   }
 
-  if (send_https_msg(ipv4, ta_port, SEND_TRANSACTION_API, req_body, seed) != SC_OK) {
+  https_response_t* response = NULL;
+  https_response_t req = {
+      .buffer = req_body,
+      .len = ret,
+  };
+
+  https_ctx_t https_ctx = {
+      .host = ipv4,
+      .port = atoi(ta_port),
+      .api = SEND_TRANSACTION_API,
+      .ssl_seed = seed,
+      .s_req = &req,
+      .s_res = response,
+  };
+
+  if (send_https_msg(&https_ctx) != SC_OK) {
     ta_log_error("http message sending error.\n");
     return SC_ENDPOINT_SEND_TRANSFER;
   }
 
+  if (response != NULL) {
+    ta_log_info("HTTP Response: %s\n", response);
```

Comment:
It does not make sense to dump the response in the end of this function. What is you intention?